integration-tests/src/test/no-proving/MicroDao.test.ts

```diff
@@ -0,0 +1,1435 @@
+  counterResetToDefault,
+  getKey,
+  historicMerkleTreeCheckRoot,
+  historicMerkleTreeInsert,
+  kernelClaimZswapCoinReceive,
+  kernelClaimZswapCoinSpend,
+  kernelClaimZswapNullfier,
+  kernelSelf,
+  merkleTreeCheckRoot,
+  merkleTreeInsert,
+  merkleTreeResetToDefault,
+  setInsert,
+  setMember,
+  setResetToDefault
+} from '@/test/utils/onchain-runtime-program-fragments';
+import {
+  ATOM_BYTES_1,
+  ATOM_BYTES_16,
+  ATOM_BYTES_32,
+  ATOM_BYTES_8,
+  ATOM_COMPRESS,
+  ATOM_FIELD,
+  EMPTY_VALUE,
+  ONE_VALUE,
+  THREE_VALUE,
+  TWO_VALUE
+} from '@/test/utils/value-alignment';
+
+describe('Ledger API - MicroDao', () => {
+  const ADVANCE = 'advance';
+  const BUY_IN = 'buyIn';
+  const CASH_OUT = 'cashOut';
+  const SET_TOPIC = 'setTopic';
+  const VOTE_COMMIT = 'voteCommit';
+  const VOTE_REVEAL = 'voteReveal';
+
+  test('should simulate DAO operations', () => {
+    const orgSk = Random.generate32Bytes();
+    const sep = [Static.encodeFromText('lares:udao:pk')];
+    const orgPk = persistentCommit([ATOM_BYTES_32], sep, [orgSk]);
+
+    const ops = setupOperations();
+
+    const state = TestState.new();
+    const REWARDS_AMOUNT = 5_000_000_000n;
+    const token: ShieldedTokenType = Random.defaultShieldedTokenType();
+
+    state.rewardsShielded(token, REWARDS_AMOUNT);
+    state.giveFeeToken(25);
+
+    const unbalancedStrictness = new WellFormedStrictness();
+    unbalancedStrictness.enforceBalancing = false;
+    const balancedStrictness = new WellFormedStrictness();
+
+    const fundsBefore = Array.from(state.zswap.coins)
+      .filter((c) => c.type === token.raw)
+      .map((c) => c.value)
+      .reduce((acc, val) => acc + val, 0n);
+
+    const { addr, encodedAddr } = part1Deploy({
+      state,
+      orgPk,
+      ops,
+      unbalancedStrictness,
+      balancedStrictness
+    });
+
+    part2SetTopic({
+      state,
+      addr,
+      orgSk,
+      orgPk,
+      setTopicOp: ops.setTopicOp,
+      unbalancedStrictness,
+      balancedStrictness
+    });
+
+    const partSks = [Random.generate32Bytes(), Random.generate32Bytes()];
+    const partPks = [
+      persistentCommit([ATOM_BYTES_32], sep, [partSks[0]]),
+      persistentCommit([ATOM_BYTES_32], sep, [partSks[1]])
+    ];
+    const partNames = ['red', 'blue'];
+
+    part3BuyIn({
+      state,
+      addr,
+      encodedAddr,
+      token,
+      partSks,
+      partPks,
+      partNames,
+      buyInOp: ops.buyInOp,
+      unbalancedStrictness,
+      balancedStrictness
+    });
+
+    const partVotes = [true, false];
+    part4VoteCommitment({
+      state,
+      addr,
+      partSks,
+      partPks,
+      partVotes,
+      partNames,
+      voteCommitOp: ops.voteCommitOp,
+      unbalancedStrictness,
+      balancedStrictness
+    });
+
+    part5AdvanceToReveal({
+      state,
+      addr,
+      orgSk,
+      orgPk,
+      advanceOp: ops.advanceOp,
+      unbalancedStrictness,
+      balancedStrictness
+    });
+
+    part6VoteRevealPhase({
```

Comment:
I would rather name it like:
```
// part 6
voteRevealPhase(...)
```
adding that "partX" prefix will quickly bite you once you decide to add one more action in between